hooks/usePoH.ts

```diff
@@ -54,7 +54,8 @@ export const usePoHs = (roles: {
     [addr: string]: boolean;
   };
 } => {
-  const addresses = Object.keys(roles);
+  const addresses = useMemo(() => Object.keys(roles), [roles]);
```

Comment:
@vidvidvid this is the main change that fixed the infinite re-render

Reply:
aaaah, nice one. thank you for fixing it! :)